nominal/nominal.py

```diff
@@ -355,20 +357,20 @@ def __getattr__(self, k: str) -> None:
 
     def __init__(
         self,
-        rid: str = None,
-        path: str = None,
-        paths: list[str] = [],
-        datasets: list[Dataset] = [],
-        properties: dict = {},
-        title: str = None,
+        rid: str | None = None,
+        path: str | None = None,
+        paths: Sequence[str] = (),
+        datasets: Sequence[Dataset] | Mapping[str, Dataset] = (),
+        properties: dict[str, str] | None = None,
```

Comment:
we should prefer the generic containers (e.g. `Mapping` over `dict`, `Sequence` over `list`) in the client signatures to be more flexible for consumers. However I only updated the ones in this PR that I was actively changing - we can do that in a later cleanup PR